Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,264 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+import Foundation
+
+/// ViewController for the site and facility prompt.
+class SiteFacilityPromptViewController: UIViewController {
+    weak var delegate: FloorFilterViewControllerDelegate?
+    var viewModel = FloorFilterViewModel()
+    
+    /// UI Elements and constraints
+    @IBOutlet weak var siteFacilitySearchBar: UISearchBar!
+    @IBOutlet weak var backBtn: UIButton!
+    @IBOutlet weak var closeBtn: UIButton!
+    @IBOutlet weak var promptTitle: UILabel!
+    @IBOutlet weak var promptSubtitle: UILabel!
+    @IBOutlet weak var promptTitleSubtitleStackView: UIStackView!
+    @IBOutlet weak var siteFacilityTableView: UITableView!
+    @IBOutlet weak var bottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var designableViewHeight: NSLayoutConstraint!
+    
+    private var originalYPositionForPanel: CGFloat = 0.0
+    
+    /// Show the facilities list directly if the map has no sites configured or if there is a previously selected facility.
+    private var isShowingFacilities = false
+    private var isSearchActive = false
+    
+    /// Filtered facilities and sites list based on search query.
+    private var filteredSearchFacilities: [AGSFloorFacility] = []
+    private var filteredSearchSites: [AGSFloorSite] = []
+    
+    private var selectedSiteIndexPath = IndexPath(row: 0, section: 0)
+    
+    private func filteredFacilities() -> [AGSFloorFacility] {
+        return isSearchActive ? filteredSearchFacilities : viewModel.facilities
+    }
+    
+    private func filteredSites() -> [AGSFloorSite] {
+        return isSearchActive ? filteredSearchSites : viewModel.sites
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        isShowingFacilities = (viewModel.sites.isEmpty || viewModel.selectedFacility != nil)
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        initializeSiteFacilityTableView()
+        initializeSiteFacilitySearchBar()
+        initializeButtonsClickListeners()
+       
+        updatePromptViewHeight()
+        updatePromptTitle()
+    }
+    
+    private func initializeButtonsClickListeners() {
+        closeBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.closeSiteFacilityPrompt)))
+        closeBtn.isUserInteractionEnabled = true
+        backBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.backButtonPressed)))
+        backBtn.isUserInteractionEnabled = true
+    }
+    
+    private func updatePromptViewHeight() {
+        self.designableViewHeight.constant = 380
```

Comment:
Since this constraint is coming from the storyboard, I'm curious why this constant isn't just set there?